src/Compilers/CSharp/Portable/SourceGeneration/CSharpGeneratorDriver.cs

```diff
@@ -29,6 +39,25 @@ private CSharpGeneratorDriver(GeneratorDriverState state)
         {
         }
 
+        /// <summary>
+        /// Creates a new instance of <see cref="CSharpGeneratorDriver"/> with the specified <see cref="ISourceGenerator"/>s and default options
+        /// </summary>
+        /// <param name="generators">The generators to create this driver with</param>
+        /// <returns>A new <see cref="CSharpGeneratorDriver"/> instance.</returns>
+        public static CSharpGeneratorDriver Create(params ISourceGenerator[] generators)
```

Comment:
Thank you, this PR fixes #46847  <3

Reply:
Meant to reference that! Hopefully this way we can make the unit tests easier for people.

Actually, it's got me thinking if we should make the constructor internal completely: that way we have more freedom to change things down the line.

Reply:
The only problematic parameter is the `AnalyzerConfigOptionsProvider`, and I'm guessing there isn't a known scenario where a 3rd party would provide their own implementation of that?

I would think a `public CSharpGeneratorDriver(CSharpParseOptions parseOptions, ImmutableArray<ISourceGenerator> generators, ImmutableArray<AdditionalText> additionalTexts)` would be useful to _someone_ though.

Reply:
It seems to me that, with these public APIs, there's no real need to make the constructors public.